langchain/memory/summary.py

```diff
@@ -48,13 +48,18 @@ def from_messages(
         chat_memory: BaseChatMessageHistory,
         *,
         summarize_step: int = 2,
+        existing_summary: str = "",
```

Comment:
currently someone could already pass in buffer as part of **kwargs and that'd be passed in at instantiation, but then overwritten by either existing_summary or messages, which is kinda confusing.

what if we did
```python
  @classmethod
  def from_messages(
      cls,
      llm: BaseLanguageModel,
       chat_memory: BaseChatMessageHistory,
       *,
       summarize_step: int = 2,
       buffer: str = "",
       **kwargs:  Any
   ) -> ConversationSummaryMemory:
     obj = cls(llm=llm, chat_memory=chat_memory, buffer=buffer, **kwargs)
     if not buffer:
       for i in range(0, len()...)):
         obj.buffer = obj.predict_new_summary(...)
     return obj

Reply:
@baskaryan Makes sense! I updated the PR to use the buffer arg.

Reply:
@baskaryan Resolved this